### PR TITLE
Popover: Deprecating showcaret and color=blue

### PR DESCRIPTION
**Pull request by eniomoura**
#### What changed?

Renamed the Popover showcaret prop and the color blue for Popover, to allow for deprecation of remaining Popover instances in Pinboard still using those.

#### Why?

The remaining deprecation cases may break instances of Popover in Pinboard if migrated straight to PopoverEducational, so we decided to mark these for deprecation in Pinboard instead, and keep the remaining Popovers unchanged for now, while still allowing deprecation of the component.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-6752)

### Checklist

- [ Yes ] Updated unit and Flow Tests
- [ Yes ] Updated documentation + accessibility tests
- [ Yes ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: packages/gestalt/src/Popover.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ export default function Popover({
   color = 'white',
   role = 'dialog',
   shouldFocus = true,
-  showCaret = false,
+  _deprecatedShowCaret,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover.js:98]: you removed the `= false`

Comment (AlbertCarreras) [reply packages/gestalt/src/Popover.js:98]: this is prob failing the snapshot. i'd bring it back

**Review by AlbertCarreras — APPROVED** [review]
R+ with that default prop change pending revert

File: packages/gestalt/src/Popover.js (hunk 1 above)
Comment (eniomoura) [reply packages/gestalt/src/Popover.js]: DONE

**Pull request merged**